os/arch/arm/src/amebasmart/amebasmart_usb.c

```diff
@@ -800,4 +816,53 @@ void usb_initialize(void)
 	}
 }
 
-
+/****************************************************************************
+ * Name: usb_up_lowputc
+ *
+ * Description:
+ *   Output one packet on the serial console
+ *
+ * Input Parameters:
+ *   *ch - chatacter buffer to output
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+void usb_up_lowputc(uint8_t *ch, uint16_t len)
+{
+	uint16_t remain_data_len = len;
+	uint16_t send_data_len = 0;
+	uint8_t *buffer = NULL;
+	buffer = (uint8_t *)rtw_zmalloc(CONFIG_CDC_ACM_BULK_IN_XFER_SIZE);
+	if (buffer == NULL) {
+		udbg("malloc failed\n");
+		return;
+	}
+	while (remain_data_len > 0) {
+		if (remain_data_len >= CONFIG_CDC_ACM_BULK_IN_XFER_SIZE) {
+			send_data_len = CONFIG_CDC_ACM_BULK_IN_XFER_SIZE;
+		} else {
+			send_data_len = remain_data_len;
+		}
+		memcpy(buffer, ch, send_data_len);
+		int ret;
+		do {
+			ret = sem_wait(&g_usbdev.exclsem);
```

Comment:
we can't use sem wait during low level log outputting,
because it is used in ISR and fault handler

Reply:
Hi @ewoodev,
Do you have any suggestion for this? We need to ensure that there will only be one place calling usbd_cdc_acm_transmit

Reply:
We need more lower layer api. 
I think, we can't use usbd_cdc_acm_transmit.

Could you please create new api which is lower than usbd_cdc_acm_transmit?
we can't use sem, so, we need to use busy wait in most low usb cdc acm. and it should not wait to complate of normall usb transmit as usbd_cdc_acm_transmit
is it possible?